server.js

```diff
@@ -1,39 +1,114 @@
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from "express";
+import cors from "cors";
+import mongoose, { mongo } from "mongoose";
+import dotenv from "dotenv";
+import listEndpoints from "express-list-endpoints";
+
+import booksData from "./data/books.json";
+
+dotenv.config();
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const bookSchema = new mongoose.Schema({
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number,
+});
+
+const Book = mongoose.model("Book", bookSchema);
+
+// RESET_DB=true npm run dev
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    await Book.deleteMany();
+
+    booksData.forEach(async (item) => {
+      const newBook = new Book(item);
+      await newBook.save();
+    });
+  };
+  seedDB();
+}
+
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(express.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+app.get("/books", async (req, res) => {
+  const { author, title, page = 0 } = req.query;
+  const authorRegex = new RegExp(author, "i");
+  const titleRegex = new RegExp(title, "i");
+
+  try {
+    const books = await Book.find({
+      authors: authorRegex,
+      title: titleRegex,
+    })
+      .skip(page * 10)
+      .limit(10);
+
+    res.status(200).json(books);
+  } catch (error) {
+    res.status(400).json({ error: "Something went wrong", details: error });
+  }
+});
+
+app.get("/books/book/:bookID", async (req, res) => {
+  const { bookID } = req.params;
+
+  try {
+    const singleBook = await Book.findById(bookID);
+    if (singleBook) {
+      res.status(200).json(singleBook);
+    } else {
+      res.status(404).json({ error: "No books with this ID" });
+    }
+  } catch (error) {
+    res.status(400).json({ error: "Something went wrong", details: error });
```

Comment:
This error line is used a lot, it could be an idea to put it in a function and call it when needed ^^